cli/src/client/mod.rs

```diff
@@ -0,0 +1,44 @@
+use graphql_client::{GraphQLQuery, Response};
+use http::Uri;
+use reqwest::blocking;
+use std::error::Error;
+
+mod queries;
+
+static PROD_GQL_API_URL: &str = "https://engine-graphql.apollographql.com/api/graphql";
+
+pub struct Client {
+    api_key: String,
+    uri: Uri,
+    reqwest: blocking::Client,
+}
+
+impl Client {
+    fn from(api_key: String) -> Client {
+        Client {
+            api_key,
+            uri: PROD_GQL_API_URL.parse::<Uri>().unwrap(),
+            reqwest: blocking::Client::new(),
+        }
+    }
+
+    fn send<Q: GraphQLQuery>(
+        &self,
+        variables: Q::Variables,
+    ) -> Result<Response<Q::ResponseData>, Box<dyn Error>> {
+        let request_body = Q::build_query(variables);
+
+        let res = self
+            .reqwest
+            .post(&self.uri.to_string())
+            .header("Content-Type", "application/json")
+            .header("apollographql-client-name", "internal-apollo-cli")
+            .header("apollographql-client-version", "0.1")
+            .header("x-api-key", &self.api_key)
+            .json(&request_body)
+            .send()?;
+
+        let response_body: Response<Q::ResponseData> = res.json()?;
+        Ok(response_body)
```

Comment:
One thing this doesn't do is handle GraphQL errors (i.e. `{ errors }`). What is the type of `ResponseData` here?

Reply:
`response_body` is of type:
```rust
pub struct Response<Data> {
    /// The absent, partial or complete response data.
    pub data: Option<Data>,
    /// The top-level errors returned by the server.
    pub errors: Option<Vec<Error>>,
}
```

`ResponseData` is codegen'd based on the query, in our case it's a struct with a single `service` field.

Reply:
These are queries are here for example purposes, we should definitely error check once we have real queries (and then we can delete these)